src/components/AnswerForm/index.js

```diff
@@ -1,61 +1,162 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
 import { useParams } from 'react-router-dom';
 
-import { createAnswer } from '@api';
+import { createAnswer, getProductById } from '@api';
 
-const AnswerForm = ({ question_id }) => {
-  const [formData, setFormData] = useState({});
+import Button from '@components/Button';
+import { StyledForm, StyledFormRow, StyledLabel } from './styles';
 
-  const doHandleChange = (e) => {
-    setFormData({
-      ...formData,
-      [e.target.name]: e.target.value,
-    });
-  };
+const AnswerFrom = ({ question, setIsShowing }) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm();
+  const { id } = useParams();
+  const [productName, setProductName] = useState('');
+
+  useEffect(async () => {
```

Comment:
`useEffect` takes in only synchronous functions, but as you have here, you may write `async` functions within it